src/Featurizers/Components/WindowFeaturizerBase.h

```diff
@@ -299,6 +301,18 @@ CircularBuffer<T>::CircularBuffer(size_t max_size) : _max_size(max_size), _start
     _data.reserve(_max_size);
 }
 
+template <class T>
+void CircularBuffer<T>::clear() {
+    _data.clear();
+
+    // Make sure that the start offset is back to 0;
+    _start_offset = 0;
+    
+    // The c++ reference says "the vector capacity is not guaranteed to change due to calling this function."
+    // Since the capacity may change, calling reserve again just to make sure we still have the correct size.
+    _data.reserve(_max_size);
+}
+
```

Comment:
CircularBuffer constructor could call this method to remove some duplicate code.

Reply:
Per off line discussion I have removed the duplicate code and am calling clear from the constructor.